common/app/services/fronts/FrontJsonFapi.scala

```diff
@@ -1,70 +1,66 @@
   val bucketLocation: String
-  val parallelJsonPresses = 32
-  val futureSemaphore = new FutureSemaphore(parallelJsonPresses)
 
-  def blockingOperations: BlockingOperations
+  private def s3ObjectIdFor(path: String, prefix: String): ObjectId =
+    ObjectId(
+      S3.bucket,
+      s"$bucketLocation/${path.replaceAll("""\+""", "%2B")}/fapi/pressed.v2$prefix.json",
+    )
 
-  private def getAddressForPath(path: String, prefix: String): String =
-    s"$bucketLocation/${path.replaceAll("""\+""", "%2B")}/fapi/pressed.v2$prefix.json"
-
-  def get(path: String, pageType: PressedPageType)(implicit
-      executionContext: ExecutionContext,
-  ): Future[Option[PressedPage]] =
-    errorLoggingF(s"FrontJsonFapi.get $path") {
-      pressedPageFromS3(getAddressForPath(path, pageType.suffix))
-    }
-
-  private def parsePressedPage(
-      jsonStringOpt: Option[String],
-  )(implicit executionContext: ExecutionContext): Future[Option[PressedPage]] =
-    futureSemaphore.execute {
-      blockingOperations.executeBlocking {
-        jsonStringOpt.map { jsonString =>
-          DurationMetric.withMetrics(FaciaPressMetrics.FrontDecodingLatency) {
-            // This operation is run in the thread pool since it is very CPU intensive
-            Json.parse(jsonString).as[PressedPage]
-          }
+  private val pressedPageCache: ETagCache[ObjectId, PressedPage] = new ETagCache(
+    S3ObjectFetching(S3Async, Bytes)
+      .timing(
+        successWith = FrontDownloadLatency.recordDuration,
+        notModifiedWith = FrontNotModifiedDownloadLatency.recordDuration,
+      )
+      .thenParsing { bytes =>
+        withMetrics(FrontDecodingLatency) {
+          Using(new GZIPInputStream(bytes.asInputStream()))(Json.parse(_).as[PressedPage]).get
         }
-      }
-    }
+      },
+    AlwaysWaitForRefreshedValue,
+    _.maximumSize(180).expireAfterAccess(1.hour),
+  )
 
-  private def loadPressedPageFromS3(path: String) =
-    blockingOperations.executeBlocking {
-      DurationMetric.withMetrics(FaciaPressMetrics.FrontDownloadLatency) {
-        S3.getGzipped(path)
+  def get(path: String, pageType: PressedPageType)(implicit ec: ExecutionContext): Future[Option[PressedPage]] =
+    errorLoggingF(s"FrontJsonFapi.get $path") {
+      val objectId = s3ObjectIdFor(path, pageType.suffix)
+      pressedPageCache.get(objectId).map(Some(_)).recover {
+        case s3Exception: S3Exception =>
+          logS3ExceptionWithDevHint(objectId, s3Exception)
+          None
       }
     }
-
-  private def pressedPageFromS3(
-      path: String,
-  )(implicit executionContext: ExecutionContext): Future[Option[PressedPage]] =
-    errorLoggingF(s"FrontJsonFapi.pressedPageFromS3 $path") {
-      for {
-        s3FrontData <- loadPressedPageFromS3(path)
-        pressedPage <- parsePressedPage(s3FrontData)
-      } yield pressedPage
-    }
-
 }
 
-class FrontJsonFapiLive(val blockingOperations: BlockingOperations) extends FrontJsonFapi {
```

Comment:
Since we're deleting `BlockingOperations` [this config](https://github.com/guardian/frontend/blob/main/facia/conf/application.conf#L27-L32) can also go.

Reply:
Ah, that bit of config is still used by the `BlockingOperations` class, and that hasn't been deleted (unfortunately?!) - we're no longer using it for the Fronts-JSON download code, but it's still used elsewhere, eg [`dfp.OrderAgent`](https://github.com/guardian/frontend/blob/08dabf7b8cb2ed8a53b249be9551ddf6407ff3b4/admin/app/dfp/OrderAgent.scala#L9).